src/autoLayer.js

```diff
@@ -0,0 +1,78 @@
+gisportal.autoLayer = {};
+
+// Gets the value of a url parameter
+gisportal.autoLayer.url_get = function(param){
+   
+   var sPageURL=decodeURIComponent(window.location.search.substring(1)),
+      sURLVariables=sPageURL.split('&'),
+      sParameterName,
+      i;
+
+   for (i=0; i < sURLVariables.length; i++) {
+      sParameterName=sURLVariables[i].split('=');
+
+      if (sParameterName.length >= 0 &&sParameterName[0] === param){
+         return sParameterName[1] === undefined ? true : sParameterName[1];
+      }
+   }
+};
+
+// This function decides either to load a single layer or to refine the panel to show a list of matching layers
+gisportal.autoLayer.loadGivenLayer = function(){
+   given_wms_url = gisportal.autoLayer.url_get('wms_url');
+   if(given_wms_url){
+      given_wms_url = given_wms_url.split("?")[0];
+   }
+   given_url_name = gisportal.autoLayer.url_get('url_name');
+
+   // This passes the variables given in the URL to the getLayers function to get the matching layer(s)
+   gisportal.given_layers = gisportal.autoLayer.getLayers(given_wms_url, given_url_name);
+
+   // If there is a single layer, then it is loaded.
+   if(_.size(gisportal.given_layers) == 1){
+      try{
+         gisportal.refinePanel.layerFound(_.values(gisportal.given_layers)[0].id);
+      }
+      catch(e){
+         console.log("There was an error adding the layer with ID " + given_layers[0].id + " : " + e);
```

Comment:
Have a look at showing this error to the users by tapping into gisportal.gritter.showNotification(), and the console.log below